docs/collator/SetupAndRun/keys.md

```diff
@@ -51,78 +52,131 @@ Starting with v3.2.1 you must provide all 3 of the following keys, earlier revis
     --scheme sr25519 \
     --network calamari \
     --output-type json \
     --words 12 \
-    > ./nimbus.json
+    > ./nmbs.json
+  echo '{
+    "jsonrpc":"2.0",
+    "id":1,
+    "method":"author_insertKey",
+    "params": [
+      "nmbs",
+      "nmbs mnemonic phrase",
+      "nmbs hex public key"
+    ]
+  }' | jq \
+    --arg mnemonic "$(jq -r .secretPhrase ./nmbs.json)" \
+    --arg hex "$(jq -r .publicKey ./nmbs.json)" \
+    '. | .params[1] = $mnemonic | .params[2] = $hex' > ./insert-nmbs.json
+  echo '{
+    "jsonrpc":"2.0",
+    "id":1,
+    "method":"author_hasKey",
+    "params": [
+      "nmbs hex public key",
+      "nmbs"
+    ]
+  }' | jq \
+    --arg hex "$(jq -r .publicKey ./nmbs.json)" \
+    '. | .params[0] = $hex' > ./check-nmbs.json
   ```
-- Generate a VRF key with subkey
+- Generate a VRF (rand) key and insert/check payloads with subkey/jq
   ```bash
   #!/bin/bash
   subkey generate \
     --scheme sr25519 \
     --network calamari \
     --output-type json \
     --words 12 \
-    > ./vrf.json
-  ```
-- Install the [jq utility](https://stedolan.github.io/jq/download/) for your platform 
-- Create an `author_insertKey` RPC payload
-  ```bash
-#!/bin/bash
-echo '{
+    > ./rand.json
+  echo '{
     "jsonrpc":"2.0",
     "id":1,
     "method":"author_insertKey",
     "params": [
-      "aura",
-      "<mnemonic phrase1>",
-      "<public key1>",
-      "nmbs",
-      "<mnemonic phrase2>",
-      "<public key2>",
       "rand",
-      "<mnemonic phrase3>",
-      "<public key3>"
+      "rand mnemonic phrase",
+      "rand hex public key"
+    ]
+  }' | jq \
+    --arg mnemonic "$(jq -r .secretPhrase ./rand.json)" \
+    --arg hex "$(jq -r .publicKey ./rand.json)" \
+    '. | .params[1] = $mnemonic | .params[2] = $hex' > ./insert-rand.json
+  echo '{
+    "jsonrpc":"2.0",
+    "id":1,
+    "method":"author_hasKey",
+    "params": [
+      "rand hex public key",
+      "rand"
     ]
   }' | jq \
-    --arg s_aura "$(jq -r .secretPhrase ./aura.json)" \
-    --arg p_aura "$(jq -r .publicKey ./aura.json)" \
-    --arg s_nimbus "$(jq -r .secretPhrase ./nimbus.json)" \
-    --arg p_nimbus "$(jq -r .publicKey ./nimbus.json)" \
-    --arg s_vrf "$(jq -r .secretPhrase ./vrf.json)" \
-    --arg p_vrf "$(jq -r .publicKey ./vrf.json)" \
-    '. | .params[1] = $s_aura | .params[2] = $p_aura | .params[4] = $s_nimbus | .params[5] = $p_nimbus | .params[7] = $s_vrf | .params[8] = $p_vrf' > ./payload.json
+    --arg hex "$(jq -r .publicKey ./rand.json)" \
+    '. | .params[0] = $hex' > ./check-rand.json
   ```
-- Execute the `author_insertKey` RPC payload
+- Execute the `author_insertKey` RPC payloads
   ```bash
   #!/bin/bash
-  curl \
-    --header 'Content-Type: application/json;charset=utf-8' \
-    --data @./payload.json \
-    http://localhost:9133
+  for key in aura nmbs rand; do
+    curl \
+      --header 'Content-Type: application/json;charset=utf-8' \
+      --data @./insert-${key}.json \
+      http://localhost:9133
```

Comment:
I think this needs a note to find your RPC port from the node logs if it's different

Reply:
i agree 100%, but that is beyond the scope of this particular pr where the remit is to correct the nimbus key insertion instructions. if we allow the scope of this pr to creep so that it includes ws port number determinations, we have to consider some new and unrelated issues:
- did the reader install a distribution package? if so, the port number can be grep'd from the systemd unit file
- is the reader using docker? if so the port number is likely the substrate default.
- is the reader using a binary and a custom unit file? if so, is the unit file using one of the convention-based paths (`/etc/systemd/system/calamari.service`, `/usr/lib/systemd/system/calamari.service`), which one? 

all of this is worth documenting well, but doing so is beyond the scope of this patch.